common/errors.go

```diff
@@ -1524,12 +1526,22 @@ type ErrEndpointClientSideException struct{ BaseError }
 
 const ErrCodeEndpointClientSideException = "ErrEndpointClientSideException"
 
-var NewErrEndpointClientSideException = func(cause error) error {
+func NewErrEndpointClientSideException(cause error, retryableTowardNetwork ...bool) error {
```

Comment:
How about making this an array of flags/options?

```suggestion
func NewErrEndpointClientSideException(cause error, opts ...ExceptionOption) error {
```

then usage will be: 
```golang
common.NewErrEndpointClientSideException(cause, RetryableTowardNetwork)
```

Reply:
ah gotcha. so more explicit and also if we want to have other flags such as `RetryableTowardUpstream`, we can have that as well.

Reply:
Exactly. with the logic that we have in `isRetryableTowardNetwork`, if we add it explicitly to any of the exceptions details param; we would explicitly set the retyable behavior.